Inventory.java

```diff
@@ -0,0 +1,45 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/*
+*  Inventory.java sets the carSeatCovers list and adds a car into the list. Creates a list for themes.
+* method for comparing the user entry to the products in the inventory.txt file */
+
+public class Inventory {
+
+    // field
+    private final List<CarSeatCover> carSeatCovers = new ArrayList<>();
+
+    // method to add a car seat to the above field
+    public void addCarSeatCover(CarSeatCover carseatcover) {this.carSeatCovers.add(carseatcover);}
+
+    /* Adding the themes to a set. Called back in getUserCriteria */
+    public Set<String> getAllThemes(){
+        Set<String> allThemes = new HashSet<>();
+        for (CarSeatCover c: carSeatCovers){
+            allThemes.add(c.getTheme());
+        }
+        return allThemes;
+    }
+
+    /* Method to find a matching car seat cover between the user entry and what is stored in inventory.txt*/
+    public List<CarSeatCover> findMatch(CarSeatCover dreamProduct) {
```

Comment:
the assignment might have told you to do it this way, but in reality, this would be done a bit differently.

Along with removing the min and max prices, and the userType from the CarSeatCover class, I would probably just turn those 3 fields into parameters in this method. 

In my mind, if I ask the inventory manager to "find me a match" in the warehouse, and I give them the object I want them to find, I expect to be handed _one exact match_. 

On the other hand, if I ask someone to find me a match, but I give them only a price range and a desired type, then I would expect to find potentially multiple matches. 

To reiterate, the `CarSeatCover` object shouldn't really care about what min or max price you are searching for, it should only tell you what it's own properties are. However, if you say to the inventory manager, "Hey, can you find me some CarSeatCovers?", they'll ask you back, "Sure, what are you looking for" and that's when you give them a price range and a type.

Hope this helps :/

Reply:
The resulting function might look something like this 

```
public List<CarSeatCover> findMatch(float minPrice, float maxPrice, Type desiredType, string theme) {
    List<CarSeatCover> productMatches = new ArrayList<>();
    for (CarSeatCover c: this.carSeatCovers) {

        // early exit if product not in price range
        if (c.getPrice() < minPrice 
            || c.getPrice() > maxPrice {
             continue;
        }

        // early exit if product doesn't match the theme
        if (c.getTheme != theme)) continue;

        // early exit if product doesn't have a type that was passed in
         if (!c.getAllTypes().contains(desiredType)) continue;

        // we can now be sure that the product is within the price range, 
        // the themes match and the desiredType is included in the 
        // product's types. At this point we are confident we have a match
        productMatches.add(c);
    }

    return productMatches;
}
```